cmd/samples/samples.go

```diff
@@ -0,0 +1,213 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samples
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+
+	"github.com/apigee/apigee-remote-service-cli/cmd/provision"
+	"github.com/apigee/apigee-remote-service-cli/shared"
+	"github.com/apigee/apigee-remote-service-cli/templates"
+	"github.com/apigee/apigee-remote-service-envoy/server"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	nativeTemplateDir = "native"
+	istioTemplateDir  = "istio-1.6"
+)
+
+type samples struct {
+	*shared.RootArgs
+	isNative      bool
+	outDir        string
+	overwrite     bool
+	RuntimeHost   string
+	TargetService targetService
+	TLS           tls
+	EncodedName   string
+}
+
+type targetService struct {
+	Name   string
+	Host   string
+	Prefix string
+}
+
+type tls struct {
+	Dir string
+	Key string
+	Crt string
+}
+
+// Cmd returns base command
+func Cmd(rootArgs *shared.RootArgs, printf shared.FormatFn) *cobra.Command {
+	s := &samples{
+		RootArgs:      rootArgs,
+		TargetService: targetService{},
+		TLS:           tls{},
+	}
+
+	c := &cobra.Command{
+		Use:   "samples",
+		Short: "Managing sample configuration files for remote-service deployment",
+		Long:  `Managing sample configuration files for remote-service deployment`,
+		Args:  cobra.NoArgs,
+	}
+
+	c.AddCommand(cmdCreateSampleConfig(s, printf))
+
+	return c
+}
+
+func cmdCreateSampleConfig(s *samples, printf shared.FormatFn) *cobra.Command {
+	c := &cobra.Command{
+		Use:   "create",
+		Short: "Create sample configuration files for native envoy or istio",
+		Long: `Create sample configuration files for native envoy or istio. A valid config
+yaml file generated through provisioning is required via --config/-c. Files will be in
+the directory specified via --out (default ./samples).
+In the case of native envoy, it takes the host of the target service, the desired name
+for its cluster and optionally the path prefix for matching. It also sets custom SSL 
+connection from the envoy to the remote-service cluster if a folder containing tls.key
+and tls.crt is provided via --tls/-t.
+In the case of istio where envoy proxy acts as sidecars, if the target is unspecified,
+the httpbin example will be generated. Otherwise, users are responsible for preparing
+files related to deployment of their target services.`,
+		Args: cobra.NoArgs,
+
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			err := s.loadConfig()
+			if err != nil {
+				return errors.Wrap(err, "loading config yaml file")
+			}
+			return errors.Wrap(s.createSampleConfigs(printf), "creating sample config files")
```

Comment:
We should emit a final message when it finishes successfully. Maybe remind them of the next steps: label the default namespace and run kubectl apply on the directory?

Reply:
Sure.